scalableminds/webknossos
# Improve error logging for bucket requests

- Changes the error handling in most of DatasetArray and the related classes to Fox instead of exceptions
 - Separates typing of loaded chunk bytes from creating chunks from fill value
 - Adds error logging in BinaryDataService for failed bucket requests. This logging is rate-limited, only 5 errors are logged per dataset until the next reset (interval of 60  minutes) or dataset reload/cacheclear (Note that some times a bit more logging will happen since the counting is not thread-safe, but that should be ok, and avoids slow synchronization)

### Steps to test:
- Add errors in chunk loading. I tested it by inserting `throw new Exception("blosc boom!")` in Compressor.scala L266 (in BloscCompressor.decompress) and then loading a local zarr dataset
- Load data, should see some readable error logging, but not thousands of lines
- Page reload should not lead to more error logging, only dataset cache clear (use the reload button in dataset layers tab of left sidebar)

### Issues:
- contributes to #6906 
- fixes #7038 (though I changed the logic a bit from the proposed idea there)

------
